joomla/joomla-cms
# [New Feature] Multilingual: Propagating existing associations if desired

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/21250

### Summary of Changes
This adds a "Propagate" button in the Associations tab when editing a contact, category, article, menu item, newsfeed.
This button will display when an association is selected in a Content Language field, offering the option to Propagate the associations defined for that item to the other Content Languages.

The button will display ONLY when there are more than 2 Content Languages defined for the site.
A message will display with the results obtained.



It is B/C as it will not break anything and one is totally free to choose or not to use the button, thus keeping the possibility to Clear/Select/Create any associations as before.

Using it prevents breaking existing associations when the wrong items are chosen for some fields.

A big thank you to **Robbie Jackson** for the long work to get this new functionality.
Target Milestone: 3.9.0

### Testing Instructions
Create a Multilingual site with 3 or more languages.
Create items in each type: (contact, category, article, menu item, newsfeed) for 2 of these languages and associate them in each case.

Create a new item for each type, select another language than the ones which are already associated.
Save (NOT save and close) this item.
Display the Association tab.

You will see, as usual (here for 4 Content Languages)

<img width="929" alt="screen shot 2018-07-31 at 11 13 04" src="https://user-images.githubusercontent.com/869724/43450317-c48878ea-94b2-11e8-8e4c-5909de16d42e.png">

I do have already associations set for English and French.

I select for English the item which I know is associated.

The button displays (Better tip welcome)

<img width="741" alt="screen shot 2018-07-31 at 11 17 17" src="https://user-images.githubusercontent.com/869724/43450583-52b65e70-94b3-11e8-8f87-1f29fe47479b.png">


Clicking on the button will give

<img width="908" alt="screen shot 2018-07-31 at 11 18 22" src="https://user-images.githubusercontent.com/869724/43450653-773d0258-94b3-11e8-8940-deae7bacc484.png">

Result: the already associated category in French is added in the French field.
As we have no association set for Spanish, the field remains empty.
The message displays (it is the same in all cases except if there is an Ajax error)
`All existing associations have been set. If some are missing you may have to select or create them manually.`

If we have already associated all fields, clicking on Propagate will just display the message.

If we change an association in one of the fields and click Propagate, it is the associations from that language item that will be propagated and therefore replace the ones existing.

### Documentation Changes Required
Yes
